distributed_training/validator/forward.py

```diff
@@ -50,400 +37,143 @@ async def forward(self):
             f"Local Epoch {self.local_progress.epoch} Behind Global Epoch {self.global_progress.epoch}. Loading Latest Model State."
         )
         load_state_from_peer(self, epoch=self.global_progress.epoch)
 
-    # Evaluate wether to run an AllReduce or a Train synapse based
-    # on the global samples accumulated
+    # Evaluate wether to run an AllReduce or validate HF miner states
+    blocks_since_allreduce = self.block - self.last_allreduce_block
+    should_allreduce = blocks_since_allreduce >= self.config.neuron.blocks_per_allreduce
 
-    if (
-        (
-            (
-                self.config.neuron.global_batch_size_train
-                - self.global_progress.samples_accumulated
-            )
-            <= 25
-        )
-        and (self.global_progress.epoch == self.local_progress.epoch)
-        and (
-            (self.uid != self.master_uid)
-            or (self.local_progress.samples_accumulated != 0)
-        )
-    ):
+    responses = [[]]
+    self.miner_uids = []
+    if should_allreduce:
+        self.event.update({"synapse_type": "all_reduce"})
+        all_reduce = True
         if self.uid == self.master_uid:
-            # If running an AllReduce synapse, call as many miners as possible
-            sample_size = int(self.metagraph.n)
-        else:
-            sample_size = self.config.neuron.sample_size
+            # Master validator coordinates AllReduce and queries miners
+            try:
+                sample_size = int(self.metagraph.n)
 
-        all_reduce = True
-        self.event.update({"synapse_type": "all_reduce"})
+                # Get active miners
+                while len(self.miner_uids) < self.config.neuron.min_group_size:
+                    bt.logging.info(
+                        f"Found {len(self.miner_uids)} UIDs. Attempting to find {self.config.neuron.min_group_size - len(self.miner_uids)} more UIDs."
+                    )
+                    self.miner_uids = await get_random_uids(
+                        self,
+                        dendrite=self.dendrite,
+                        k=1,
+                        epoch=self.local_progress.epoch if all_reduce else None,
+                    )
 
-    elif (self.uid == self.master_uid) and (
-        self.local_progress.samples_accumulated == 0
-    ):
-        sample_size = int(self.metagraph.n)
-        all_reduce = False
-        self.event.update({"synapse_type": "train"})
+                self.peerids_to_uids = {
+                    str(value["peer_id"]): key
+                    for key, value in self.uid_metadata_tracker.items()
+                }
 
-    else:
-        # If running a Train synapse call, only call the sample_size
-        sample_size = self.config.neuron.sample_size
-        all_reduce = False
-        self.event.update({"synapse_type": "train"})
+                if not hasattr(
+                    self, "bandwidth"
+                ):  # TODO Maybe run different location to not wait on this
```

Comment:
Since we're moving any bandwidth related functions out of this release we should just remove this.